### ISSUE TITLE
"no-use-before-define" not catching any violations when env=node

### ISSUE THREAD

**Issue by FredKSchott** [eslint/eslint#2372]
Hey all, I just realized that eslint wasn't picking up a "no-use-before-define" rule violation in my codebase. I created a small test file based and eslint was still unable to catch it:

```js
// test.js
console.log(a);
var a = 'abcdefg';
```

```
$ eslint -v
v0.19.0
$ eslint --reset --rule 'no-use-before-define: 2' test.js
$ echo $?
0
```

It looks like this rule is heavily tested in eslint itself, any idea why this wouldn't be catching? All other rules seem to be working fine.

**Comment by nzakas** [comment]
I'm unable to reproduce. Is it possible you're turning off the rule in a config file or elsewhere?

**Comment by FredKSchott** [comment]
Oh interesting, I didn't realize that --reset doesn't reset any rules enabled/disabled by the `env` your .eslintrc defines. 

So it looks like it has something to do with env=node, when I remove "env=node" from my .eslintrc (or retry in my home directory without any .eslintrc) I cannot reproduce either. Are you able to reproduce these results?

```
$ eslint --reset --rule 'no-use-before-define: 2' test.js

test.js
  2:12  error  a was used before it was defined  no-use-before-define

✖ 1 problem (1 error, 0 warnings)
```
```
$ eslint --reset --rule 'no-use-before-define: 2' --env "browser" test.js

test.js
  2:12  error  a was used before it was defined  no-use-before-define

✖ 1 problem (1 error, 0 warnings)
```
```
$ eslint --reset --env "node" --rule 'no-use-before-define: 2' test.js
   // Exits successfully
```

**Comment by ilyavolodin** [comment]
Yes, you are correct. Turning on node environment seems to disable no-use-before-define, but it shouldn't.

**Comment by nzakas** [comment]
It must have something to do with the extra node.js scope.

**Comment by alphatan** [comment]
The problem was introduced after v0.10.2, and can be produced across the lastest version.

$node -v
v0.12.2

$eslint -v
v0.20.0

**Comment by FredKSchott** [comment]
Bisected this morning out of curiosity, looks like the bug was introduced in e5aea7f92bd6dfe67ca8f6943ae090a72b4115f5

**Comment by nzakas** [comment]
Yup, that's what I meant by the extra node.js scope.

**Comment by nzakas** [comment]
Working on this.

**Comment by FredKSchott** [comment]
:+1: awesome, thanks Nicholas!



—
